app/src/pages/inside/stepPage/modals/editDefectModal/editDefectModal.jsx

```diff
@@ -207,14 +171,62 @@ export class EditDefectModal extends Component {
     return getStorageItem(REPLACE_COMMENTS_TO_ALL_AVAILABILITY_NAME);
   };
 
+  getCloseConfirmationConfig = () =>
+    this.checkIfTheDataWasChanged()
+      ? {
+          confirmationWarning: this.props.intl.formatMessage(messages.warningMessage),
+        }
+      : null;
+
+  getItemsToNextAction = () =>
+    this.mapStateToFetchData().map((item) => ({ ...item, id: item.test_item_id }));
+
+  mapStateToFetchData = () => {
```

Comment:
I think something like `prepareDataToSend` is better. We do not use state here, so current name is a bit confusing

Reply:
We use state inside method, but I agree, this name is clearer.